### ISSUE TITLE
[java] AvoidCatchingGenericException: False positive when exception is rethrown

### ISSUE THREAD

**Issue by emouty** [pmd/pmd#6451]
**Affects PMD Version:** 7.21.0

**Rule:**
[AvoidCatchingGenericException](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#avoidcatchinggenericexception)

**Description:**

Sometime we need to handle rollback of a process when any kind of exception occurs. In these cases, once the rollback was handled the exception is rethrown. I feel that when the exception is rethrown after this rule shouldn't be raised. I don't know if this should be default behaviour or not but maybe at least having a parameter (something like `excludeCasesWhereExceptionIsRethrown`) on the rule that allow that would be great. 

**Code Sample demonstrating the issue:**

```java
 try {
          someProcessThatCouldFail()
        } catch (Exception e) {
           handleRollback()
            throw e;
        }
```

Other case where we group all exceptions under a new one to ease error management in api responses. 
```java
 try {
          someProcessThatCouldFail()
         } catch (Exception ex) {
            throw new IllegalStateException("Failed run process", ex);
        }
```

**Expected outcome:**

I would expect that the issue is not flagged for theses cases

**Running PMD through:** *Maven*, *Intellij plugin*
